src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -296,12 +300,48 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
                     services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, FunctionsScaleMonitorService>());
                 }
                 services.TryAddSingleton<FunctionsScaleManager>();
+
+                services.AddHostOverrides();
```

Comment:
This is adding storage specific implementations so doesn't it rely on the above call to AddAzureStorageProvider happening? That call above is conditional, which means that this could be called without the other required services being registered?

Reply:
We already call AddAzureStorageProvider in the WebHost root builder. The conditional is meant to avoid registering two instances of these services.